File: src/com/basmilius/ps/bastools/core/config/BasToolsConfigurable.java
package com.basmilius.ps.bastools.core.config;

import com.intellij.openapi.options.ConfigurationException;
import com.intellij.openapi.options.SearchableConfigurable;
import org.jetbrains.annotations.Nls;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import javax.swing.*;

public class BasToolsConfigurable implements SearchableConfigurable
{

	private BasToolsConfigurableGUI gui;

	@Nls
	@Override
	public final String getDisplayName ()
	{
		return "Bas Tools Preferences";
	}

	@NotNull
	@Override
	public final String getHelpTopic ()
	{
		return "preference.BasTools";
	}

	@NotNull
	@Override
	public final String getId ()
	{
		return "preference.BasTools";
	}

	@Nullable
	@Override
	public final Runnable enableSearch (final String searchQuery)
	{
		return null;
	}

	@Nullable
	@Override
	public final JComponent createComponent ()
	{
		this.gui = new BasToolsConfigurableGUI();
		return this.gui.getRootPanel();
	}

	@Override
	public final boolean isModified ()
	{
		return false;
	}

	@Override
	public final void apply () throws ConfigurationException
	{
	}

	@Override
	public final void reset ()
	{
	}

	@Override
	public final void disposeUIResources ()
	{
		this.gui = null;
	}

}

File: src/com/basmilius/ps/bastools/core/config/BasToolsConfigurable.kt
package com.basmilius.ps.bastools.core.config

import com.intellij.openapi.options.ConfigurationException
import com.intellij.openapi.options.SearchableConfigurable
import org.jetbrains.annotations.Nls

import javax.swing.*

/**
 * Class BasToolsConfigurable
 *
 * @author Bas Milius
 * @package com.basmilius.ps.bastools.core.config
 */
class BasToolsConfigurable : SearchableConfigurable
{

	private var gui : BasToolsConfigurableGUI? = null

	/**
	 * {@inheritdoc}
	 *
	 * @author Bas Milius
	 */
	@Nls
	override fun getDisplayName() : String
	{
		return "Bas Tools Preferences"
	}

	/**
	 * {@inheritdoc}
	 *
	 * @author Bas Milius
	 */
	override fun getHelpTopic() : String
	{
		return "preference.BasTools"
	}

	/**
	 * {@inheritdoc}
	 *
	 * @author Bas Milius
	 */
	override fun getId() : String
	{
		return "preference.BasTools"
	}

	/**
	 * {@inheritdoc}
	 *
	 * @author Bas Milius
	 */
	override fun enableSearch(searchQuery : String?) : Runnable?
	{
		return null
	}

	/**
	 * {@inheritdoc}
	 *
	 * @author Bas Milius
	 */
	override fun createComponent() : JComponent?
	{
		this.gui = BasToolsConfigurableGUI()
		return this.gui!!.rootPanel
	}

	/**
	 * {@inheritdoc}
	 *
	 * @author Bas Milius
	 */
	override fun isModified() : Boolean
	{
		return false
	}

	/**
	 * {@inheritdoc}
	 *
	 * @author Bas Milius
	 */
	@Throws(ConfigurationException::class)
	override fun apply()
	{
	}

	/**
	 * {@inheritdoc}
	 *
	 * @author Bas Milius
	 */
	override fun reset()
	{
	}

	/**
	 * {@inheritdoc}
	 *
	 * @author Bas Milius
	 */
	override fun disposeUIResources()
	{
		this.gui = null
	}

}
